### PR TITLE
Add documentation for querying S3 data with CLI

### PR DESCRIPTION
**Pull request by andygrove**
# Which issue does this PR close?

Closes https://github.com/apache/arrow-datafusion/issues/3399

 # Rationale for this change

We should tell users how to use these new features

# What changes are included in this PR?

Docs

# Are there any user-facing changes?

No

### CODE REVIEW

**Review by avantgardnerio — APPROVED** [review]
Thank you for adding this!

File: datafusion-examples/examples/query-s3.rs [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+/// This example demonstrates querying data in an S3 bucket.
+///
+/// The following environment variables must be defined:
+///
+/// - AWS_DEFAULT_REGION
+/// - AWS_ACCESS_KEY_ID
+/// - AWS_SECRET_ACCESS_KEY
+///
+#[tokio::main]
+async fn main() -> Result<()> {
```
Comment (andygrove) [inline datafusion-examples/examples/query-s3.rs:33]: This example currently fails with:

```
Error: ObjectStore(Generic { store: "S3", source: MissingLastModified })
```

Comment (andygrove) [reply datafusion-examples/examples/query-s3.rs:33]: I filed https://github.com/apache/arrow-datafusion/issues/3633 for this

File: datafusion-cli/README.md [hunk 2]
```diff
@@ -65,6 +65,37 @@ DataFusion CLI v12.0.0
 1 row in set. Query took 0.017 seconds.
 ```
 
+## Querying S3 Data Sources
+
+The CLI can query data in S3 if the following environment variables are defined:
+
+- `AWS_DEFAULT_REGION`
```
Comment (tustvold) [inline datafusion-cli/README.md:72]: AWS_REGION is more standard, AWS_DEFAULT_REGION is the fallback

File: datafusion-cli/README.md [hunk 3]
```diff
@@ -65,6 +65,41 @@ DataFusion CLI v12.0.0
...
+
+```bash
+$ aws s3 cp test.csv s3://my-bucket/
+upload: ./test.csv to s3://my-bucket/test.csv
+
+$ export AWS_REGION=us-east-1
+$ export AWS_SECRET_ACCESS_KEY=***************************
+$ export AWS_ACCESS_KEY_ID=**************
+
+$ ./target/release/datafusion-cli
```
Comment (alamb) [inline datafusion-cli/README.md:90]: this is so cool

**Review by alamb — APPROVED** [review]
Thanks @andygrove

**Pull request merged**